modules/javafx.graphics/src/main/java/com/sun/javafx/scene/text/TextLayout.java

```diff
@@ -208,11 +208,12 @@ public Hit(int charIndex, int insertionIndex, boolean leading) {
      * @param text text for which HitInfo needs to be calculated.
      *             It is expected to be null in the case of {@link javafx.scene.text.TextFlow}
      *             and non-null in the case of {@link javafx.scene.text.Text}
-     * @param textRunStart Text run start position.
-     * @param curRunStart starting position of text run where hit info is requested.
+     * @param textRunStart Start position of first Text run where hit info is requested.
+     * @param curRunStart Start position of text run where hit info is requested.
+     * @param forTextFlow Indicates if the hit info is requested for TextFlow or Text node. {@code true} for TextFlow and {@code false} for Text node.
```

Comment:
The `getHitInfo` calculates the hit test information for the Text nodes embedded inside a TextFlow as well. Because of this `boolean isThisForTextFlow = this.spans != null;` is not enough.
When hit test is requested on a Text embedded in TextFlow, if we use `getText()`, it gives the entire content rather than giving the Text node on which hit test is requested. Since we only have x, y coordinates, I found sending text as parameter is necessary.
We are sending the run start value and start value of the first run in the line, but this also did not prove to be enough when we have cases such as multiline text, emojis, RTL text embedded in LTR text, repeated texts etc. So I have kept the `text` parameter as it is and removed `forTextFlow` parameter. This is now calculated using following condition
`boolean forTextFlow = spans != null && text == null;`